lodash.js

```diff
@@ -6208,6 +6208,7 @@
      * This method is like `_.pull` except that it accepts an array of values to remove.
      *
      * **Note:** Unlike `_.difference`, this method mutates `array`.
+     * **Note:** To remove elements from an array based on a predicate, see `_.remove`.
```

Comment:
The reason for the variety of pulls is that `pull` accepts any value or values to remove. The value could be an array. So to make composition easier there's a `pullAll` which accepts an array of values to pull. This means `pull` is just `rest(pullAll)`. Sometimes you want to pull by a transform so then there's `pullAllBy`. We avoid the `pullBy` form because that would be a variable arg signature depending on how many values you want to pull. Finally `pullAt` is for removing by index and not value.